### ISSUE TITLE
PHP 8.1 notices raised by iterator proxies -> will not work in PHP>=9

### ISSUE THREAD

**Issue by mschop** [eloquent/pops#10]
See: https://php.watch/versions/8.1/internal-method-return-types

This is caused by missing return type declarations for the iterator methods.

E.g.:

```
Deprecated: Return type of Eloquent\Pops\ProxyObject::offsetExists($key) should either be compatible with ArrayAccess::offsetExists(mixed $offset): bool, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress t
he notice in /code/vendor/eloquent/pops/src/ProxyObject.php on line 152
```

**Comment by ezzatron** [comment]
Thanks for the heads up. Should be a relatively easy fix.

**Comment by ezzatron** [comment]
Fixed in [`5.0.0`](https://github.com/eloquent/pops/releases/tag/5.0.0).
